File: source/downloads/code/LeetCode/Python/Reverse-Words-in-a-String.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        ret = ""
        if not s:
            return ret

        words = s.split()
        for i in range(len(words)-1,-1,-1):
            ret = ret+words[i] + " "
        return ret[:len(ret)-1]

File: source/downloads/code/LeetCode/Java/Reverse-Words-in-a-String.java
public class Solution {
    public String reverseWords(String s) {
	String[] parts = s.trim().split("\\s+");
	String ret = "";
	for(int i=parts.length-1; i>0; i--){
	    ret += parts[i]+" ";
	}
	return ret+parts[0];
    }
}
